src/pages/home/report/ReportActionsView.js

```diff
@@ -364,7 +341,12 @@ class ReportActionsView extends React.Component {
             );
         }
 
-        this.setUpUnreadActionIndicator();
+        if (this.newMessageMarkerPosition < 0) {
+            this.newMessageMarkerPosition = this.props.report.unreadActionCount === 0
```

Comment:
I wonder if we really need this variable, but correct me if this sounds wrong...

Since the `newMessageMarkerPosition` can always be inferred by looking at `props` we can instead make a method that calculates the marker position + tells us if a current `sequenceNumber` matches? I think it would be cleaner to call a method in `renderItem` that does this so that the logic isn't all over the place. 

I think that will eliminate the need to "reset" this variable back to `-1`, but haven't tested this out.

Reply:
`newMessageMarkerPosition` can't really be always inferred by looking at `props`, right? Once we update the unreadActionCount (which gets updated when we update the last action), we'll lose that info.

Reply:
It seems there are really only two things we need to know, yet we are expressing it in a single variable (and also updating it in a render method). This makes everything a little harder to understand IMO. Hopefully this helps think about the problem in a different way.

We need to know...

1. Whether we should show a new message marker at all. something like `shouldShowNewMessageMarker` stored in state could tell us this. When would we need to hide and show this marker?
2. If we should show the new message marker what sequenceNumber should we show it at? The easier question as this can _always_ be known by looking at the props like we are already doing here.

Reply:
1. Yep, we can totally have that piece of data in a different variable or in the state if we want.
2. Nope. That's the issue I've been fighting with. `this.props.report.unreadActionCount` gets updated to zero after the 3 seconds timer ends. At that point, if we haven't saved the sequence in a variable, we lose it. When a new message comes in and the component gets rerendered, we'll lose the marker.

Reply:
Sorry for all the back and forth here. I'm having a hard time understanding things, but I think turning a corner on a real suggestion. 😅 

>we can totally have that piece of data in a different variable or in the state if we want

Ok I think actually such a variable is not necessary, but it's an interesting exercise to think about _when_ it might be `true`. 

I think it'd be `true` when our `unreadActionCount > 0` (you've got that part), _but_ we need to use the unread count when the report view mounts (set in the constructor) or we switch to a new report. 

This is where things get unclear for me...

>this.props.report.unreadActionCount gets updated to zero after the 3 seconds timer ends. At that point, if we haven't saved the sequence in a variable, we lose it. When a new message comes in and the component gets rerendered, we'll lose the marker.

Right now we are setting this to `-1` when the `reportID` changes and then updating it in the `render()` method when it's less than 0. This seems like kind of an anti-pattern because we're using `-1` as a "signal" to the `render()` method that _it_ needs to save this reference. That's not really the job of the render method and we should be able to save this reference when the component mounts or when the reportID changes.

So we should be able to end up with something like:

- Set `this.initialUnreadActionCount`  in `componentDidMount` and when we switch to a new report (`reset()`)
- Do the same checks we already have in `renderItem()`

I think that would clean things up a bit. Does that sound like it might work?

Reply:
Haha, no worries! I also thought this would be simpler than it ended up being. This didn't work for a couple of reasons:
- I can't get rid of the `-1` check (unless we use a different variable): 
When we swap reports, `componentDidUpdate` is actually called twice. Once with `prevProps.reportID !== newProps.reportID`, which is when we call reset, and once more afterwards to actually update the report contents.
On the first call, we still don't have the correct `unreadActionCount`, so I can't set the variable at that point. We need to wait till the second round, so I still need to check that "set this only once" flag (the `-1`, in this case).
- I still tried moving the logic to `componentDidUpdate` so it wouldn't be in `render`, but since this change does't trigger a rerender, even though the variable gets set correctly, the marker doesn't show up until something else happens. Maybe I could force a redraw, but we probably want to avoid that. 😢

Reply:
Ok, thanks for explaining. It seems clear what is happening here is that there is one update for the `reportID` then another update when [`withOnyx`](https://github.com/Expensify/react-native-onyx/blob/master/lib/withOnyx.js) re-fetches anything with keys depending on props (so `reportActions` and `report` in this case). 

Seems like we have two options...

1. Implement this workaround for now (maybe try to add a lot of comments to explain why we need to do what we're doing)
2. Fix Onyx so that we don't run into this situation again in the future

What would you prefer to do?

My expectation is that if a key depends on a prop and the prop it depends on changes we should check to see if we can also batch additional updates together instead of causing an extra re-render. I'm going to start looking into a solution on the Onyx side regardless and if we want to hold off for that, cool.

Otherwise, we can go ahead with the workaround, but here's what I'd suggest to make everything a little clearer...

1. In `reset()` set a flag on an instance variable called `this.didReportIDChange` to `true`
2. In `componentDidUpdate()` check to see if this is equal to `true`
3. Create a method called `this.getInitialUnreadCount()` and use is to initialize a variable in state `this.state.initialUnreadActionCount` in the constructor
4. In `componentDidUpdate()` check for `this.didReportIDChange`  and use `getInitialUnreadCount()` to update the value of `this.state.initialUnreadActionCount` and set `this.didReportIDChange` back to `false`.

This will "force" an update. But I can't say whether that is a problem or not. I'd think maybe not.